src/components/FilesTable.tsx

```diff
@@ -81,6 +85,30 @@ export const FilesTable = ({
     return [{ ...filesInfo, mainFile: true }, ...(filesInfo?.Files ?? [])].filter((file) => filterFiles(file));
   }, [sourceFilesInfo, context?.reportData.sourceFilesInfo, filterFiles]);
 
+  const processedWithIssues = React.useMemo(() => {
+    const fileDetails = fileRecords || context?.reportData.filerecords;
+
+    const fileStatusMap: { [fileId: string]: boolean } = {};
+    for (const dataRow of data) {
```

Comment:
@mayank99 Here is a quick breakdown for the logic of this:

For each row (file) in the table: 

We are first find the associated `FileRecord` for the file 
```
.filter((fd) => fd.file?.identifier == dataRow.fileId)
```
Since I was unsure of whether or not a file could have one or more associated `FileRecord`, we are taking all 
the `FileRecord`s we found, taking each one's array of `AuditRecord`s and reducing them into a single array
```
 .reduce((a: AuditRecord[], b) => a.concat(b.auditrecords || ({} as AuditRecord)), [])
 ```
 Then we are going through that flat array of `AuditRecord`s and seeing if any of their `AuditInfo` properties has a level which indicates and issue.
 
```
.some(
          (ar) =>
            ar.auditinfo?.level == 'Critical' ||
            ar.auditinfo?.level == 'Fatal' ||
            ar.auditinfo?.level == 'Warning' ||
            ar.auditinfo?.level == 'Error'
        );
```